### PR TITLE
util: fix data race in the cpu test

### LINKED ISSUE #40125: DATA RACE in the TestCPUValue
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
==================
WARNING: DATA RACE
Read at 0x00c0002615d0 by goroutine 21:
  github.com/pingcap/tidb/util/cpu.(*Observer).observe()
      util/cpu/cpu.go:94 +0x364
  github.com/pingcap/tidb/util/cpu.(*Observer).Start.func1()
      util/cpu/cpu.go:69 +0x23a
Previous write at 0x00c0002615d0 by goroutine 10:
  github.com/pingcap/tidb/util/cpu.(*Observer).observe()
      util/cpu/cpu.go:99 +0x3bc
  github.com/pingcap/tidb/util/cpu.(*Observer).Start.func1()
      util/cpu/cpu.go:69 +0x23a
Goroutine 21 (running) created at:
  github.com/pingcap/tidb/util/cpu.(*Observer).Start()
      util/cpu/cpu.go:60 +0xa6
  github.com/pingcap/tidb/util/cpu.TestCPUValue()
      util/cpu/cpu_test.go:45 +0x19b
  testing.tRunner()
      GOROOT/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      GOROOT/src/testing/testing.go:1493 +0x47
Goroutine 10 (running) created at:
  github.com/pingcap/tidb/util/cpu.(*Observer).Start()
      util/cpu/cpu.go:60 +0xa6
  github.com/pingcap/tidb/util/cpu.TestCPUValue()
      util/cpu/cpu_test.go:28 +0x4e
  testing.tRunner()
      GOROOT/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      GOROOT/src/testing/testing.go:1493 +0x47
================== 
```

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #40125

Problem Summary:

### What is changed and how it works?

wrongly call ```Observer.Start()```  twice
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/cpu/cpu_test.go [hunk 1]
```diff
@@ -25,7 +25,6 @@ import (
 
 func TestCPUValue(t *testing.T) {
 	Observer := NewCPUObserver()
```
Comment (lance6716) [inline util/cpu/cpu_test.go:27]:
```suggestion
	observer := NewCPUObserver()
```

**Review by lance6716 — APPROVED** [review]

File: util/cpu/cpu_test.go (hunk 1 above)
Comment (hawkingrei) [reply util/cpu/cpu_test.go]: Done

**Review by xhebox — APPROVED** [review]

**Pull request merged**